src/rviz/default_plugin/marker_utils.cpp

```diff
@@ -76,4 +84,346 @@ MarkerBase* createMarker(int marker_type, MarkerDisplay* owner, DisplayContext*
+{
+  if(marker.scale.x == 0.0)
+  {
+    addCommaIfRequired(ss);
+    ss << "width LINE_LIST or LINE_STRIP is 0.0 (scale.x)";
+  }
+  else if(marker.scale.y != 0.0 || marker.scale.z != 0.0)
+  {
+    addCommaIfRequired(ss);
+    ss << "scale.y and scale.z of LINE_LIST or LINE_STRIP are ignored";
+  }
+}
+
+void checkScalePoints(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.scale.x == 0.0 || marker.scale.y == 0.0)
+  {
+    addCommaIfRequired(ss);
+    ss << "width and/or height of POINTS is 0.0 (scale.x, scale.y)";
+  }
+  else if(marker.scale.z != 0.0)
+  {
+    addCommaIfRequired(ss);
+    ss << "scale.z of POINTS is ignored";
+  }
+}
+
+void checkScaleText(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.scale.z == 0.0)
+  {
+    addCommaIfRequired(ss);
+    ss << "text height of TEXT_VIEW_FACING is 0.0 (scale.z)";
+  }
+  else if(marker.scale.x != 0.0 || marker.scale.y != 0.0)
+  {
+    addCommaIfRequired(ss);
+    ss << "scale.x and scale.y of TEXT_VIEW_FACING are ignored";
+  }
+}
+
+void checkColor(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.color.a == 0.0)
+  {
+    addCommaIfRequired(ss);
+    ss << "marker is fully transparent (color.a is 0.0)";
+  }
+}
+
+void checkPointsArrow(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.points.size() != 0 && marker.points.size() != 2)
+  {
+    addCommaIfRequired(ss);
+    ss << "Number of points for an ARROW marker should be either 0 or 2";
+  }
+}
+
+void checkPointsNotEmpty(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.points.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "points should not be empty for specified marker type" ;
+  }
+  else if(marker.type == visualization_msgs::Marker::TRIANGLE_LIST && (marker.points.size() % 3) != 0)
+  {
+    addCommaIfRequired(ss);
+    ss << "number of points should be a multiple of 3 for TRIANGLE_LIST Marker";
+  }
+  else if(marker.type == visualization_msgs::Marker::LINE_LIST && (marker.points.size() % 2) != 0)
+  {
+    addCommaIfRequired(ss);
+    ss << "number of points should be a multiple of 2 for LINE_LIST Marker";
+  }
+  else if(marker.type == visualization_msgs::Marker::LINE_STRIP && marker.points.size() <= 1)
+  {
+    addCommaIfRequired(ss);
+    ss << "at least two points are required for a LINE_STRIP Marker";
+  }
+}
+
+void checkPointsEmpty(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(!marker.points.empty())
+  {
+    addCommaIfRequired(ss);
+    ss << "points array is ignored by specified marker type";
+  }
+}
+
+void checkColors(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if(marker.colors.size() == 0)
+  {
+    checkColor(marker, ss);
+  }
+
+  if(marker.colors.size() != marker.points.size())
+  {
+    addCommaIfRequired(ss);
+    ss << "number of colors is not equal to number of points or 0";
```

Comment:
The error message doesn't match the checks: marker.colors might be empty here, while marker.points are not.

Reply:
Previously there was a return before this check so it was always true that the colors array is not empty. This has been re-added in c414eac